morpho-checkpoint-node/src/Morpho/Tracing/Pretty.hs

```diff
@@ -315,71 +489,773 @@ instance
+  mpretty ConnectValencyExceeded = "Someone else manged to create the final connection to a target before us."
+
+instance MPretty DnsTrace where
+  -- DnsTrace already has a nice humain readable Show instance
   mpretty = viaShow
 
-instance MPretty (WithIPList (SubscriptionTrace Socket.SockAddr)) where
+instance MPretty Socket.SockAddr where
   mpretty = viaShow
 
-instance (Show peer) => MPretty (WithMuxBearer peer MuxTrace) where
+instance (MPretty a) => MPretty (NtN.ConnectionId a) where
+  mpretty cid = mpretty (NtN.remoteAddress cid)
+
+instance MPretty a => MPretty (WithIPList a) where
+  mpretty (WithIPList _ destAddrs event) =
+    "With destinations" <+> mpretty destAddrs <> ":" <+> mpretty event
+
+instance MPretty LocalAddress where
+  mpretty (LocalAddress fp) = "LocalAddress(" <> pretty fp <> ")"
+
+instance (MPretty peer, MPretty a) => MPretty (WithMuxBearer peer a) where
+  mpretty (WithMuxBearer peer a) = "With mux bearer" <+> mpretty peer <> ":" <+> mpretty a
+
+instance MPretty MuxTrace where
   mpretty = viaShow
 
 instance MPretty a => MPretty [a] where
-  mpretty xs = list $ map mpretty xs
+  mpretty = ppList . map mpretty
 
-instance (StandardHash blk) => MPretty (ChainHash blk) where
-  mpretty = viaShow
+instance (MPretty a, MPretty b) => MPretty (Map a b) where
+  mpretty m =
+    "("
+      <> hsep (punctuate comma (map ppTuple (Map.toList m)))
+      <> ")"
+    where
+      ppTuple (a, b) = mpretty a <+> "->" <+> mpretty b
+
+instance (MPretty a, MPretty b) => MPretty (a, b) where
+  mpretty (a, b) = "(" <> mpretty a <> "," <+> mpretty b <> ")"
+
+instance (MPretty a) => MPretty (Maybe a) where
+  mpretty (Just a) = "Just(" <> mpretty a <> ")"
+  mpretty Nothing = "Nothing"
+
+instance (MPretty a, MPretty b) => MPretty (Either a b) where
+  mpretty (Left e) = "Left(" <> mpretty e <> ")"
+  mpretty (Right v) = "Right(" <> mpretty v <> ")"
+
+instance MPretty Int where
+  mpretty = pretty
+
+instance MPretty Bool where
+  mpretty = pretty
 
 instance
-  (LedgerSupportsProtocol blk) =>
-  MPretty (ChainDB.InvalidBlockReason blk)
+  (MorphoStateDefaultConstraints h c, Signable (BftDSIGN c) (MorphoStdHeader h c)) =>
+  MPretty (ChainDB.InvalidBlockReason (MorphoBlock h c))
   where
   mpretty = viaShow
 
-instance Condense (HeaderHash blk) => MPretty (Point blk) where
-  mpretty pt =
-    case pt of
-      GenesisPoint -> "genesis (origin)"
-      BlockPoint slot h -> pretty $ condense h ++ "@" ++ condense slot
+instance MPretty (HeaderHash blk) => MPretty (Point blk) where
+  mpretty GenesisPoint = "Genesis"
+  mpretty (BlockPoint slot h) =
+    "Block(slot:" <+> mpretty slot <> ", hash:" <+> mpretty h <> ")"
+
+instance MPretty (HeaderHash blk) => MPretty (RealPoint blk) where
+  mpretty (RealPoint slot headerHash) =
+    "RealPoint(slot:" <+> mpretty slot <> ", hash:" <+> mpretty headerHash <> ")"
```

Comment:
Let's just use `Block(slot:` here, the fact that this is a `RealPoint` isn't useful.